### ISSUE TITLE
variable_name should not apply to overriding properties

### ISSUE THREAD

**Issue by erikstrottmann** [realm/SwiftLint#1169]
When I'm overriding a property, I don't have the freedom to change its name. For example, when overriding a UIKit property with a very long name, I can't avoid an error:

```swift
class MyViewController: UIViewController {
    override var preferredInterfaceOrientationForPresentation: UIInterfaceOrientation {
        return .landscapeLeft
    }
}
```

> Variable Name Violation: Variable name should be between 3 and 40 characters long: 'preferredInterfaceOrientationForPresentation' (variable_name)

It doesn't make sense for SwiftLint to ask me to change the property name. If I did, it wouldn't be overriding the superclass property anymore.
